src/test/java/io/debezium/connector/yugabytedb/YugabyteDBPublicationReplicationTest.java

```diff
@@ -0,0 +1,193 @@
+package io.debezium.connector.yugabytedb;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import io.debezium.connector.yugabytedb.common.YugabyteDBContainerTestBase;
+import io.debezium.connector.yugabytedb.common.YugabytedTestBase;
+import io.debezium.connector.yugabytedb.connection.YugabyteDBConnection;
+
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.*;
+import org.yb.client.GetDBStreamInfoResponse;
+import org.yb.client.YBClient;
+
+import io.debezium.config.Configuration;
+import io.debezium.connector.yugabytedb.transforms.YBExtractNewRecordState;
+import io.debezium.transforms.ExtractNewRecordStateConfigDefinition;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class YugabyteDBPublicationReplicationTest extends YugabytedTestBase {
+
+    public static String createPublicationForTableStatement = "CREATE PUBLICATION %s FOR TABLE %s ;";
+    public static String createPublicationForALLTablesStatement = "CREATE PUBLICATION %s FOR ALL TABLES ;";
+    public static String createReplicationSlotStatement = "SELECT pg_create_logical_replication_slot('test_replication_slot', 'yboutput');";
+    public static String dropReplicationSlotStatement = "SELECT pg_drop_replication_slot('test_replication_slot');";
```

Comment:
If the above statements are generic and will be useful in other classes as well, we can put them to `TestHelper` as well.